gbpservice/nfp/modules/configurator.py

```diff
@@ -0,0 +1,458 @@
+    agent object with core service controller.
+
+    :param sc: Service Controller object that is used for interfacing
+    with core service controller.
+    :param cm: Configurator module object that is used for accessing
+    ConfiguratorModule class methods.
+    :param conf: Configuration object that is used for configuration
+    parameter access.
+    :param demuxer: De-multiplexer object that is used for accessing
+    ConfiguratorDemuxer class methods.
+
+    Returns: None
+
+    """
+
+    # Initializes RPC client
+    rpc_mgr = ConfiguratorRpcManager(sc, cm, conf, demuxer)
+    configurator_agent = rpc.RpcAgent(sc,
+                                      topic=CONFIGURATOR_RPC_TOPIC,
+                                      manager=rpc_mgr)
+
+    # Registers RPC client object with core service controller
+    sc.register_rpc_agents([configurator_agent])
+
+
+def get_configurator_module_instance(sc):
+    """ Provides ConfiguratorModule class object and loads service agents.
+
+    Returns: Instance of ConfiguratorModule class
+
+    """
+
+    cm = ConfiguratorModule(sc)
+    conf_utils = utils.ConfiguratorUtils()
+
+    # Loads all the service agents under AGENT_PKG module path
+    cm.imported_service_agents = conf_utils.load_agents(AGENTS_PKG)
+    msg = ("Configurator loaded service agents from %s location."
+           % (cm.imported_service_agents))
+    LOG.info(msg)
+    return cm
+
+
+def module_init(sc, conf):
+    """Initializes configurator module.
+
+    Creates de-multiplexer object and invokes all the agent entry point
+    functions. Initializes oslo RPC client for receiving messages from
+    REST server. Exceptions are raised to parent function for all types
+    of failures.
+
+    :param sc: Service Controller object that is used for interfacing
+    with core service controller.
+    :param conf: Configuration object that is used for configuration
+    parameter access.
+
+    Returns: None
+    Raises: Generic exception including error message
+
+    """
+
+    # Create configurator module and de-multiplexer objects
+    try:
+        cm = get_configurator_module_instance(sc)
+        demuxer_instance = demuxer.ConfiguratorDemuxer()
+    except Exception as err:
+        msg = ("Failed to initialize configurator de-multiplexer. %s."
+               % (str(err).capitalize()))
+        LOG.error(msg)
+        raise Exception(err)
+    else:
+        LOG.info("Initialized configurator de-multiplexer.")
+
+    # Initialize all the pre-loaded service agents
+    try:
+        cm.init_service_agents(sc, conf)
+    except Exception as err:
+        msg = ("Failed to initialize configurator agent modules. %s."
+               % (str(err).capitalize()))
+        LOG.error(msg)
+        raise Exception(err)
+    else:
+        LOG.info("Initialized configurator agents.")
+
+    # Initialize RPC client for receiving messages from REST server
+    try:
+        init_rpc(sc, cm, conf, demuxer_instance)
```

Comment:
Unlike in the case of lifecycle_manager/modules, there is no event_init here. Are all the event_init calls in the service_agents?